FIAT/discontinuous_lagrange.py

```diff
@@ -5,7 +5,110 @@
 #
 # SPDX-License-Identifier:    LGPL-3.0-or-later
 
+import numpy as np
 from FIAT import finite_element, polynomial_set, dual_set, functional, P0
+from FIAT.finite_element import make_barycentric_lattice_coordinates, make_index_permutations
+
+
+def make_entity_permutations(dim, npoints):
+    if npoints <= 0:
+        return {o: [] for o in range(np.math.factorial(dim + 1))}
+    # DG nodes are numbered, in order of significance,
+    # - by g0: entity dim (vertices first, then edges, then ...)
+    # - by g1: entity ids (DoFs on entities of smaller ids first)
+    # - lexicographically as in CG
+    #
+    # Example:
+    # dim = 2, degree = 3 (npoints = degree + 1)
+    #
+    #     facet ids    CG node numbers  DG node numbers
+    #    +
+    #    | \           3                1
+    #    |   \  0      2 6              8 3
+    #  2 |     \       1 5 8            7 9 4
+    #    |       \     0 4 7 9          0 5 6 2
+    #    +--------+
+    #        1
+    a = make_barycentric_lattice_coordinates(dim, npoints - 1)
+    # a = [[0, 0, 3],
+    #      [0, 1, 2],    (0,3,0)
+    #      [0, 2, 1],
+    #      [0, 3, 0],    (0,2,1)(1,2,0)
+    #      [1, 0, 2],
+    #      [1, 1, 1],    (0,1,2)(1,1,1)(2,1,0)
+    #      [1, 2, 0],
+    #      [2, 0, 1],    (0,0,3)(1,0,2)(2,0,1)(3,0,0)
+    #      [2, 1, 0],
+    #      [3, 0, 0]]    Lattice points that a represents
+    # a.shape[0] = number of DoFs
+    # a.shape[1] = dim + 1
+    # sum of each row = degree (bary centric lattice coordinates)
+    index_perms = make_index_permutations(dim + 1)
+    # index_perms = [[0, 1, 2],
+    #                [0, 2, 1],
+    #                [1, 0, 2],
+    #                [1, 2, 0],
+    #                [2, 0, 1],
+    #                [2, 1, 0]]
```

Comment:
These bits just look like commented out code, can we have some exposition?

Reply:
I think it's now easier to read.